### ISSUE TITLE
BeforeLeaveEvent::postpone() method doesn't prevent to update the browser URL

### ISSUE THREAD

**Issue by benno-mueller-saxess** [vaadin/flow#9551]
### Description of the bug / feature

After a navigation is postponed in a **BeforeLeaveListener** oder **BeforeLeaveObserver**, the url in browser changes while the view does not change (same issue as https://github.com/vaadin/flow/issues/3619).

### Minimal reproducible example

```
@Route(value = "hello", layout = MainView.class)
@PageTitle("Hello World")
@RouteAlias(value = "", layout = MainView.class)
public class HelloWorldView extends Div implements BeforeLeaveObserver {

    @Override
    public void beforeLeave(final BeforeLeaveEvent beforeLeaveEvent) {
           beforeLeaveEvent.postpone();
    }
}
```
or Example in zip below:

1. Go to About View with the menu
2. Go to Hello World View with the menu
3. Select CheckBox
4. Navigate back using browser
-> URL changes to **_/about_**

### Expected behavior

The URL should stay the same when navigation is postponed.

### Versions:
    - Vaadin / Flow version: 14.4.2 + 14.4.4
    - Java version: 8 + 11
    - OS version: Ubuntu 18.04.3 LTS
    - Browser version (if applicable): Chromium 86.0.4240.198
    - Application Server (if applicable): Tomcat
    - IDE (if applicable): IntelliJ

**Comment by benno-mueller-saxess** [comment]
[bug-postpone.zip](https://github.com/vaadin/flow/files/5623687/bug-postpone.zip)

**Comment by denis-anisimov** [comment]
This issue _is not the same_ as #3619.
The scenario is similar indeed but the important difference is : navigation is done via **BACK** browser button whereas the original report in #3619 is about navigation inside the Flow app using router link.

The difference is dramatic.
The usecase from this ticket is described here: https://github.com/vaadin/flow/issues/3619#issuecomment-613389430 (third item).

The #3619 in its original description is fixed : the same scenario but with  _back_ button replaced to click to "About" link in the app doesn't update the URL. And this is exactly the case of #3619.

I don't know the exact details of fix provided for #3619. I recall there was some talks that _back_ button is not possible to fix since this is done by the browser itself. But I'm not sure.
I will ask someone who recall the details.
May be @pleku , @bogdanudrescu could help here.

Anyway : this is not a duplicate of #3619 and it might be we may not do anything here (so the ticket will be just closed).

**Comment by pleku** [comment]
As per comment https://github.com/vaadin/flow/issues/3619#issuecomment-613389430, this issue is the part 3) of that comment and probably not fixed in it. So this is issue is very much valid 
>when the user does back/forward navigation and causes a popstatevent which is sent to server side, the URL has changed before we can affect it. The event cannot be canceled for obvious reasons. This is problematic and for the postpone case it is unknown if we can fix this. Maybe we could try to push back the old state with history.pushState() (?) if the user has triggered the postpone action from server side, but this needs to be investigated.

So we need to investigate whether or not it is possible to trigger a push state from server side to prevent back/forward navigation from the user when the server side code does a `event.postpone();`. It might be indeed that this is impossible, but lets see.

**Comment by aley2003** [comment]
As an additional side effect of postpone() the history is broken after ContinueNavigationAction.proceed().
If you postpone a back() navigation and then proceed it, a forward() is not possible as expected. 
The source location has been removed from history by postpone().
